crates/solvers/src/boundary/liquidity/stable.rs

```diff
@@ -0,0 +1,65 @@
+pub use shared::sources::balancer_v2::pool_fetching::StablePool as Pool;
+use {
+    crate::domain::{eth, liquidity},
+    ethereum_types::{H160, H256, U256},
+    shared::sources::balancer_v2::{
+        pool_fetching::{AmplificationParameter, CommonPoolState, TokenState},
+        swap::fixed_point::Bfp,
+    },
+    std::collections::BTreeMap,
+};
+
+/// Converts a domain pool into a [`shared`] Balancer V2 stable pool. Returns
+/// `None` if the domain pool cannot be represented as a boundary pool.
+pub fn to_boundary_pool(address: H160, pool: &liquidity::stable::Pool) -> Option<Pool> {
+    // NOTE: this is only used for encoding and not for solving, so it OK to
+    // use this an approximate value for now. In fact, Balancer V2 pool IDs
+    // are `pool address || pool kind || pool index`, so this approximation is
+    // pretty good.
+    let id = {
+        let mut buf = [0_u8; 32];
+        buf[..20].copy_from_slice(address.as_bytes());
+        H256(buf)
+    };
+
+    let swap_fee = to_fixed_point(&pool.fee)?;
+    let reserves = pool
+        .reserves
+        .iter()
+        .map(|reserve| {
+            Some((
+                reserve.asset.token.0,
+                TokenState {
+                    balance: reserve.asset.amount,
+                    scaling_exponent: reserve.scale.exponent(),
+                },
+            ))
+        })
+        .collect::<Option<BTreeMap<_, _>>>()?;
+    let amplification_parameter = AmplificationParameter::new(
+        *pool.amplification_parameter.numer(),
+        *pool.amplification_parameter.denom(),
+    )
+    .ok()?;
+
+    Some(Pool {
+        common: CommonPoolState {
+            id,
+            address,
+            swap_fee,
+            paused: false,
```

Comment:
Is this guaranteed because we only fetch unpaused pools?

Reply:
Yes - these are the pools that have already been fetched sent to the `baseline` solver. This "boundary" module is just mapping the pools that we get for solving into the `shared` crate types to make use of the shared code (even if we don't use all fields like this one). Additionally, the `shared::baseline_solver` code never actually checks this value.